File: Algorithms/970.Powerful-Integers/powerful_integers.go
import "math"
func powerfulIntegers(x int, y int, bound int) []int {
    var res []int
    if bound <= 1 {
        return res
    }
    lgx := 0
    if x > 2 {
        lgx = int(math.Log(float64(bound)) / math.Log(float64(x))) + 1
    } else {
        lgx = int(math.Log(float64(bound)) / math.Log(float64(2))) + 1
    }
    lgy := 0
    if y > 2 {
        lgy = int(math.Log(float64(bound)) / math.Log(float64(y))) + 1
    } else {
        lgy = int(math.Log(float64(bound)) / math.Log(float64(2))) + 1
    }
    
    i := 0
    resMap := make(map[int]bool)
    fmt.Println(lgx,lgy)
    for i < lgx {
        j := 0
        for j < lgy {
            ssum := int(math.Pow(float64(x), float64(i)) + math.Pow(float64(y), float64(j)))
            if ssum > bound {
                break
            }
            resMap[ssum] = true
            j++
        }
        i++
    }
    for k,_ := range resMap {
        res = append(res, k)
    }
    return res
}

File: Algorithms/970.Powerful-Integers/powerful_integers.py
from math import log
class Solution:
    def powerfulIntegers(self, x: int, y: int, bound: int) -> List[int]:
        res = set()
        if bound <= 1:
            return []
        lgx = int(log(bound, max(x, 2))) + 1
        lgy = int(log(bound, max(y, 2))) + 1
        for i in range(lgx):
            for j in range(lgy):
                ssum = x ** i + y ** j
                if ssum > bound:
                    break
                res.add(ssum)
        return list(res)

